crates/biome_service/src/settings.rs

```diff
@@ -220,6 +220,37 @@ impl Settings {
         result
     }
 
+    /// Return plugins taking overrides into account
+    pub fn as_plugins(&self, path: &Utf8Path) -> Cow<Plugins> {
+        let mut result = Cow::Borrowed(&self.plugins);
+
+        for pattern in &self.override_settings.patterns {
+            if pattern.is_file_included(path) {
+                result.to_mut().0.extend_from_slice(&pattern.plugins.0);
```

Comment:
Maybe we should have a `Deref` implementation on `Plugins`? Not a fan of those `.0` accesses, as it obscures what exactly is being accessed.